### PR TITLE
date range filter added

### PR DESCRIPTION
**Pull request by ankitgoyalgloballogic**
#REF: #MOBILEAPPS-1300

### CODE REVIEW

File: ContentApp/BusinessLayer/Localization/en.lproj/Localizable.strings [hunk 1]
```diff
@@ -302,7 +302,7 @@
 "low" = "Low";
 "medium" = "Medium";
 "high" = "High";
-"task-service-not-configured-message" = "This account is not configured on the process services. Please contact to administrator.";
```
Comment (aman-alfresco) [inline ContentApp/BusinessLayer/Localization/en.lproj/Localizable.strings:305]: Have you fixed the string in other languages as well?

File: ContentApp/Modules/Advance Search/Calendar/SearchCalendarComponentViewModel.swift [hunk 2]
```diff
@@ -59,13 +59,27 @@ class SearchCalendarComponentViewModel: NSObject {
         return dateFormatter.string(from: date)
     }
     
-    func date(from dateString: String) -> Date? {
+    func date(from dateString: String, format: String? = nil) -> Date? {
+        var dateFormatt = dateFormat
```
Comment (aman-alfresco) [inline ContentApp/Modules/Advance Search/Calendar/SearchCalendarComponentViewModel.swift:63]: Please rename the variable dateFormatt.

File: ContentApp/PresentationLayer/Tasks/Task List/Controller/TasksListViewController.swift [hunk 3]
```diff
@@ -299,7 +304,41 @@ extension TasksListViewController {
             sortFilterView.buildDataSource()
             self.view.addSubview(sortFilterView)
             self.sortFilterView = sortFilterView
+            sortFilterView.callBack = { (type: ComponentType?, value: [String], isReset: Bool) in
+                if isReset {
+                    self.resetComponents()
+                } else if type == .createdDateRange {
```
Comment (aman-alfresco) [inline ContentApp/PresentationLayer/Tasks/Task List/Controller/TasksListViewController.swift:310]: Can we manage these conditions in the switch case?

File: ContentApp/BusinessLayer/Localization/en.lproj/Localizable.strings (hunk 1 above)
Comment (ankitgoyalgloballogic) [reply ContentApp/BusinessLayer/Localization/en.lproj/Localizable.strings:305]: Not yet. Localization team will update this.

File: ContentApp/Modules/Advance Search/Calendar/SearchCalendarComponentViewModel.swift (hunk 2 above)
Comment (ankitgoyalgloballogic) [reply ContentApp/Modules/Advance Search/Calendar/SearchCalendarComponentViewModel.swift:63]: Done

File: ContentApp/PresentationLayer/Tasks/Task List/Controller/TasksListViewController.swift (hunk 3 above)
Comment (ankitgoyalgloballogic) [reply ContentApp/PresentationLayer/Tasks/Task List/Controller/TasksListViewController.swift]: Done

**Review by gl-lovekesh — APPROVED** [review]

**Pull request merged**